feat(backend): Use cert-manager for cache-server certificate

surajkota opened the pull request:
**Description of your changes:**
- Manifests for using cert-manager for the certificate required by cache-server
  - Created a cert-manager directory under env instead of adding the manifests under aws since this issue was [also found on Mercedes Benz distribution of Kubernetes](https://github.com/kubeflow/manifests/issues/2165#issuecomment-1096633299). This means with the current implementation(which does not use cert-manager), caching will not work on any k8s distribution which issues certificates for CSRs with signerName `kubernetes.io/kubelet-serving` only to kubelets
  - Addressed comments from #7779. Will update 7779 once this is merged.
- cherry-pick of #7538 to this(release-1.8) branch (since it will be required for 1.8.2 release)

**Testing:**
- Manually tested both standalone installation and multi user with Kubeflow installed with a custom built cache-server image since there are changes to `backend/src/cache/main.go`

surajkota commented:
@zijianjoy Please review this PR which addresses comments from #7779. We can update 7779 after this is merged with the required changes for master branch

This PR is on 1.8 release branch so it will fastrack 1.8.2 release

surajkota commented:
/test kubeflow-pipeline-e2e-test

surajkota commented:
/retest-required

zijianjoy commented:
Hello @surajkota ,

1. We only accept cherry-pick from `master` branch to `release-x.x` branch. We need to wait for https://github.com/kubeflow/pipelines/pull/7779 to be merged first. Can you check with Karthik about this? If Karthik is not going to work on this PR, you can create a PR to continue his effort for merging to master.
2. We cannot accept the current changes in `dev` folder: You can consider creating a new env folder which introduces cert-manager on top of `dev`. 
3. In the future, I hope that https://github.com/kubernetes/kubernetes/issues/63732 can introduce the new signerName which fits in our use case.

surajkota commented:
/retest

surajkota commented:
/test ?

surajkota commented:
/rerun cla

surajkota commented:
/check-cla

surajkota commented:
/retest-required

surajkota commented:
1. Makes sense. I updated this PR to be on master branch and updated the PR description accordingly. There were some issues with CI last night which seem to have resolved now. Some test failures seem to be flaky and unrelated to changes in this PR.
2. > We cannot accept the current changes in dev folder: 
   - Can you expand a bit more on this? I did not understand this comment. All the overlays in this PR are under `manifests/kustomize/env/cert-manager`. corresponding overlay for `dev` is under path `manifests/kustomize/env/cert-manager/dev` similar to `manifests/kustomize/env/cert-manager/platform-agnostic-multi-user`. I have not modified `manifests/kustomize/env/dev`
3. +1

@zijianjoy Thanks for the review. Please take a look at the latest revision

surajkota commented:
/close in favour of #7843
